packages/libs/error-stack/src/report.rs

```diff
@@ -633,10 +727,23 @@ impl<Context> std::process::Termination for Report<Context> {
     }
 }
 
-pub struct ReportImpl {
-    pub(super) frame: Frame,
-    #[cfg(all(nightly, feature = "std"))]
-    backtrace: Option<Backtrace>,
-    #[cfg(feature = "spantrace")]
-    span_trace: Option<SpanTrace>,
+impl<Context> FromIterator<Report<Context>> for Option<Report<Context>> {
+    fn from_iter<T: IntoIterator<Item = Report<Context>>>(iter: T) -> Self {
+        let mut iter = iter.into_iter();
+
+        let mut base = iter.next()?;
+        for rest in iter {
+            base.extend_one(rest);
+        }
+
+        Some(base)
+    }
+}
+
+impl<Context> Extend<Self> for Report<Context> {
+    fn extend<T: IntoIterator<Item = Self>>(&mut self, iter: T) {
+        for item in iter {
+            self.extend_one(item);
+        }
+    }
```

Comment:
Very sad we can't implement `Extend<Frame>` without allowing a `Report` to be extended from any `Report` (without matching context). Then this would have been more intuitive. 😞

Reply:
Definitely 😞